chipsec/modules/common/rom_armor.py

```diff
@@ -0,0 +1,88 @@
+# CHIPSEC: Platform Security Assessment Framework
+# Copyright (c) 2024, AMD Corporation
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; Version 2.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# Contact information:
+# chipsec@amd.com
+#
+
+
+"""
+This module verifies support for Rom Armor and SPI ROM protections.
+
+Reference:
+
+
+usage:
+    ``chipsec_main -m common.rom_armor``
+
+Examples:
+    >>> chipsec_main.py -m common.rom_armor
+
+"""
+
+from chipsec.module_common import BaseModule, MTAG_BIOS, MTAG_SMM
+from chipsec.library.returncode import ModuleResult
+from typing import List
+
+TAGS = [MTAG_BIOS, MTAG_SMM]
+
+SMU_PSP_SMN_BASE = 0x3800000
+SMU_PSP_MBOX_CMD_STATUS = 0x00010970
+
+class rom_armor(BaseModule):
+
+    def __init__(self):
+        BaseModule.__init__(self)
+
+    def is_supported(self) -> bool:
+        return self.cs.is_amd()
+
+    def check_ROMAMOR(self) -> bool:
+        reg_value = self.cs.psp.smu_read32(SMU_PSP_SMN_BASE + SMU_PSP_MBOX_CMD_STATUS)
+        self.logger.log_information(f"PSP Mailbox Status 0x{reg_value:X}")
+        reg_value = self.cs.psp.smu_read32(SMU_PSP_SMN_BASE + 0x109fc)
+        self.logger.log_information(f"PSP Mailbox Features 0x{reg_value:X}")
+        hsti = self.cs.psp.query_HSTI()
```

Comment:
A few FYIs.
1. There is no guarantee that the command exists on newer/all platforms.  The kernel only uses this command on platforms that don't report the security reporting capability bit.
2. There is a lot of other interesting information in this return.  The kernel code has offsets documented that you may want to look at.